app/services/host_counter.rb

```diff
@@ -3,12 +3,18 @@ def initialize(association)
     @association = association
   end
 
+  delegate :fetch, to: :hosts_count
```

Comment:
This allows directly using the ids when needed, and also maintains the previous interface for application controller so plugins won't break.